sled-agent/src/bootstrap/trust_quorum/config.rs

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::{Deserialize, Serialize};
+use serde_json;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+use vsss_rs::Share;
+
+use super::rack_secret::{RackSecret, Verifier};
+use crate::bootstrap::agent::BootstrapError;
+
+const FILENAME: &'static str = "trust-quorum-config.json";
+
+// This is a short term mechanism for initializing sleds with their own shares,
+// verifiers, and ancillary information.
+//
+// It is expected to be removed once rack initialization is done and the
+// RackSecret is generated dyanmically and shares and verifiers transferred over
+// SPDM channels.
+//
+// Users should only manually modify the `sled_index` and `enabled` fields.
```

Comment:
This was pretty much all demo code expected to be removed in the follow up commit to this. I actually thought this was basically the path we agreed upon for that, although it may be a tad uglier than you expected due to the reasons you have stated here.

The fact that this code is expected to be removed is why I was so concerned last night and today about being able to remotely deploy different configurations to different nodes for testing. I think having that ability basically aligns with what you are suggesting. 

I certainly do not want to check in secrets, and force enable through the manual editing of a file. That is a hideous hack that I expect to be rid of by the end of the week. However, doing it this way means that I didn't have to generate a file on one node, and copy it to another and then edit it for a quick demo. I specifically wanted to do the `omicron-package install` to show things were "real".

I'm certainly open to changing how this is done, and was about to start doing that tonight even if you didn't point out the obviously grotesque, disillusioning code here 😃 .

Rather than write up the current situation where developers currently generate this file on their own manually, I'd prefer to just write up a tool for deploying remotely and document that. If you are uncomfortable enough to merge this in as is, then I will work on that, and make it so that only if the tool is used and the user explicitly opts in will they get the trust quorum enabled. I'll put the minimal part of that tool in this commit and document it, removing all this other stuff. The enablement of trust quorum will be based on the file existing as you suggest, and not an `enable` field inside the file. 

However, I also expect to just add the protocol to generate the key dynamically from a sled and properly distribute the shares very soon. In this case, there will be no initial file on any of the nodes, and so there will need to be some sort of flag set as part of configuration in order to enable the trust quorum.

The main reason I opened the PR in this state was to generate smaller commits. I'm trying to do this all piecemeal without a minimum of new features. Since the rest of the code is coming so soon to clean this up in a much better way than config files, I didn't want to spend too much time making the hack nice.

Reply:
yeah, that makes sense. I 100% get that the path to progress is a balancing act, and feel free to push back on me.

> Rather than write up the current situation where developers currently generate this file on their own manually, I'd prefer to just write up a tool for deploying remotely and document that.

This sounds great - I believe this would likely remove the checked-in secrets, as well as the manual file editing currently necessary.

> However, I also expect to just add the protocol to generate the key dynamically from a sled and properly distribute the shares very soon. In this case, there will be no initial file on any of the nodes, and so there will need to be some sort of flag set as part of configuration in order to enable the trust quorum.

Yeah, I expect the config that would be hardcoded here is "am I a leader" or something, right?

> The main reason I opened the PR in this state was to generate smaller commits. I'm trying to do this all piecemeal with a minimum of new features per PR. Since the rest of the code is coming so soon to clean this up in a much better way than config files, I didn't want to spend too much time making the hack nice.

Makes sense. This is related to my earlier comment - push back on me if the "right way" to do things make my comments irrelevant, but I figured I'd highlight stuff that even has the potential to be a concern. I wanna put it on your radar, but ultimately, I trust your judgement here.

Reply:
>    Yeah, I expect the config that would be hardcoded here is "am I a leader" or something, right?

Probably. It will also serve as an implicit enable of trust quorum if given, because we'll need to know to not wait for share distribution on the non-leader nodes when we are using single agent mode. In the hopefully not to far future, we can just do this dynamically via a post to an HTTP endpoint or something. Discussion coming soon to a huddle near you.

Reply:
> Makes sense. This is related to my earlier comment - push back on me if the "right way" to do things make my comments irrelevant, but I figured I'd highlight stuff that even has the potential to be a concern. I wanna put it on your radar, but ultimately, I trust your judgement here.

Thanks @smklein. I appreciate the thoroughness of the review. It's great to have someone else see the same problems I see in this code, and know that we are aligned on solutions. It's always worth it to comment because there are plenty of times when I miss opportunities to improve code over time or do something in a more idiomatic manner. In this case, I am just very aware of exactly what I did wrong haha.